kv/memberlist/mergeable.go

```diff
@@ -9,6 +9,9 @@ type Mergeable interface {
 	// If merge doesn't result in any change, returns nil.
 	// Error can be returned if merging with given 'other' value is not possible.
 	//
+	// Implementations may modify the internal state even if a change is not returned,
+	// so long as the logical value does not change, i.e. the properties below are maintained.
+	//
```

Comment:
This should be moved after "result" definition, one paragraph below).

```suggestion
	// Memberlist-based KV store will keep the result even if Merge returned no change. Implementations should
	// be careful about not changing logical value when returning emtpy change.
	//
```